src/js/AppStoreMenu.js

```diff
@@ -0,0 +1,146 @@
+                <div className="svg-wrap" onClick={this.props.onSelect}>
+                    <StaticIcon class="static-icon" image="0x0F" />
+                </div>
+            </div>
+        )
+    }
+}
+
+class AppStoreMenu extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            confirmID: null,
+            appName: null,
+            appIcon: null,
+            appDescription: null
+        }
+
+        this.cancel = this.cancel.bind(this);
+        this.confirm = this.confirm.bind(this);
+    }
+
+    onSelection(appID, name, icon, description) {
+        this.setState((state, props) => {
+            return {
+                confirmID: appID,
+                appName: name,
+                appIcon: icon,
+                appDescription: description
+            }
+        });
+    }
+
+    cancel() {
+        this.setState((state, props) => {
+            state.confirmID = null;
+            return state;
+        });
+    }
+
+    confirm() {
+        fileSystemController.subscribeToEvent('UninstallApp', (success, params) => {
+            if (!success || !params.policyAppID) {
+                console.error('error encountered when uninstalling app');
+                return;
+            }
+
+            store.dispatch(addAppPendingSetAppProperties({'policyAppID': params.policyAppID}, false));
+            BCController.setAppProperties({ 
+                policyAppID: params.policyAppID,
+                enabled: false
+            });
+        });
+
+        fileSystemController.sendJSONMessage({
+            method: 'UninstallApp',
+            params: {
+                policyAppID: this.state.confirmID
+            }
+        });
+
+        this.setState((state, props) => {
+            state.confirmID = null;
+            return state;
+        });
+    }
+
+    render() {
+        const postFixes = [ "KB", "MB", "GB" ]
+        const postFixSizes = [ 1024, 1048576, 1073741824 ]
+
+        const themeClass = this.props.theme ? 'dark-theme' : 'light-theme';
+        var modalClass = themeClass + " alertOverlay"
+
+        return (
+            <div>
+                <Modal
+                isOpen={!!this.state.confirmID}
+                className="alertModal app-body"
+                overlayClassName={modalClass}
+                contentLabel="Example Modal">
+                    <ConfirmAlert appID={this.state.confirmID} name={`Uninstall ${this.state.appName}?`}
+                        description={this.state.appDescription} iconUrl={this.state.appIcon}
+                        leftText="Cancel" leftCallback={this.cancel} rightText="Uninstall" rightCallback={this.confirm} />
+                </Modal>
+                <AppHeader icon='false' backLink="/appstore" menuName="STORE"/>
+                <div className="appstore-menu">
+                    {this.props.apps ? this.props.apps.map((app) => {
+
+                        var postFixIndex = 0;
```

Comment:
Add a comment that this code is for calculating the app size that should be displayed

Reply:
added in https://github.com/smartdevicelink/generic_hmi/pull/240/commits/b79ab73fcfd760a30437c25d53d1c6665ee29526

Reply:
Also could you move the data size calculation outside of the return line. As a general rule I think for style reasons any logic or calculation that is slightly complicated should be calculated before the return.

Reply:
Because the calculation is for each menu item, maybe the calculation should be moved to the AppStoreMenuItem render() method